java/java.source/test/unit/src/org/netbeans/modules/java/JavaNodeTest.java

```diff
@@ -40,4 +48,30 @@ public void testSheetToArrayReturnsModifiableArray() {
         assertNull(s.toArray()[0]);
     }
     
+    public void testSingleJavaSourceRun() {
+        try {
+            List<String> commandsList = new ArrayList<>();
+            if (Utilities.isUnix()) {
+                commandsList.add("bash");
+                commandsList.add("-c");
+            }
+            File f1 = new File(new File(new File (getDataDir().getAbsolutePath()), "files"), "TestSingleJavaFile.java");
+            File javaPathFile = new File(new File(new File(System.getProperty("java.home")), "bin"), "java");
+            commandsList.add(javaPathFile.getAbsolutePath() + " " + f1.getAbsolutePath());
+            ProcessBuilder pb = new ProcessBuilder(commandsList);
```

Comment:
...rather than testing `ProcessBuilder` I suggest to test the code extracted from the `invokeAction`.